### PR TITLE
Add test for the generator

### LINKED ISSUE #2: Add tests for LabelProviderSnippet
- [x] Check other popular yeoman generators and if / how they implemented tests
 - [x] Implement test for LabelProviderSnippet
 - [x] Implement test for existing snippets

### PR DESCRIPTION
**Pull request by TheZoker**
Fixes #2
@sdirix

### CODE REVIEW

File: src/app/index.ts [hunk 1]
```diff
@@ -102,6 +103,11 @@ module.exports = class TheiaExtension extends Base {
             type: String,
             default: '2.4.0'
         });
+        this.option('skipInstall', {
+            description: 'Skip the dependency installation (for tests)',
```
Comment (sdirix) [inline src/app/index.ts]: This option also makes sense for normal use cases. So please change the description to `Skip install after generation`

File: src/app/index.ts [hunk 2]
```diff
@@ -291,7 +297,9 @@ module.exports = class TheiaExtension extends Base {
     }
 
     install() {
-        this.spawnCommand('yarn', []);
+        if(!this.options.skipInstall) {
```
Comment (sdirix) [inline src/app/index.ts]: check formatting

File: test/generator-test.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+            })
+            .toPromise().then(function () {
+                try {
+                    assert.file([
+                        'package.json',
+                        'README.md',
+                        `${name}/src/browser/${name}-contribution.ts`,
+                        `${name}/src/browser/${name}-frontend-module.ts`,
+                        `${name}/src/browser/style/example.css`,
+                        `${name}/src/browser/style/baseline_code_black_18dp.png`,
```
Comment (sdirix) [inline test/generator-test.js:87]: Also check for the other image

**Review by sdirix — CHANGES_REQUESTED** [review]
The tests work fine!

Please also commit the updated yarn.lock and rebase/chery-pick once the base PR is updated.

**Comment by TheZoker** [comment]
Implemented all the requests. Thanks!

**Comment by sdirix** [comment]
Please add the updated yarn lock ;)

**Comment by TheZoker** [comment]
Forgot that, sorry! Done now :)

**Comment by sdirix** [comment]
The lock is missing the test dependencies

**Comment by TheZoker** [comment]
Hmm that's the file I get when I remove `yarn.lock` and run `yarn`. I'm not sure, why this is not included...

**Comment by sdirix** [comment]
The test dependencies are added in the second commit, while the lock is updated in the first.

**Comment by TheZoker** [comment]
Actually I created the yarn lock file with the lastest commit and added it back to the first commit 😅

**Review by sdirix — APPROVED** [review]
Thanks for the info. This was my mistake, my local repo still had the previous state of the branch. Works fine for me now! Thanks!

**Comment by sdirix** [comment]
The skipInstall parameter was truthy by default, fixed with #8

**Pull request merged**